### ISSUE TITLE
Request add support for pmvhaven.com

### ISSUE THREAD

**Issue by phuisc1892367** [yt-dlp/yt-dlp#9831]
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I *intentionally* remove or skip any mandatory\* field

### Checklist

- [X] I'm reporting a new site support request
- [X] I've verified that I have **updated yt-dlp to nightly or master** ([update instructions](https://github.com/yt-dlp/yt-dlp#update-channels))
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've checked that none of provided URLs [violate any copyrights](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy) or contain any [DRM](https://en.wikipedia.org/wiki/Digital_rights_management) to the best of my knowledge
- [X] I've searched [known issues](https://github.com/yt-dlp/yt-dlp/issues/3766) and the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [ ] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and am willing to share it if required

### Example URLs

https://pmvhaven.com/video/MARAVILLOSA-PMV-_66195f01d0f2168854325fd0

### Provide a description that is worded well enough to be understood

Request for support to be added for pmvhaven.com, currently certain fields are wrong or unsupported. E.g. uploader is set to pmvhaven.com, date is NA and ID is different from url ID.

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [ ] If using API, add `'verbose': True` to `YoutubeDL` params instead
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
PS C:\Users\USERNAME> yt-dlp -vU https://pmvhaven.com/video/MARAVILLOSA-PMV-_66195f01d0f2168854325fd0
[debug] Command-line config: ['-vU', 'https://pmvhaven.com/video/MARAVILLOSA-PMV-_66195f01d0f2168854325fd0']
[debug] User config "C:\Users\USERNAME\AppData\Roaming\yt-dlp\config": ['--merge-output-format', 'mkv', '--no-mtime', '--windows-filenames', '--concurrent-fragments', '4', '--retries', '3', '-o', '[%(uploader)s] %(upload_date>%Y-%m-%d)s %(title).100s [%(id)s].%(ext)s']
[debug] Encodings: locale cp1252, fs utf-8, pref cp1252, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version nightly@2024.04.28.232723 from yt-dlp/yt-dlp-nightly-builds [ac817bc83] (win_exe)
[debug] Python 3.8.10 (CPython AMD64 64bit) - Windows-10-10.0.22631-SP0 (OpenSSL 1.1.1k  25 Mar 2021)
[debug] exe versions: ffmpeg 7.0-full_build-www.gyan.dev (setts), ffprobe 7.0-full_build-www.gyan.dev
[debug] Optional libraries: Cryptodome-3.20.0, brotli-1.1.0, certifi-2024.02.02, curl_cffi-0.5.10, mutagen-1.47.0, requests-2.31.0, sqlite3-3.35.5, urllib3-2.2.1, websockets-12.0
[debug] Proxy map: {}
[debug] Request Handlers: urllib, requests, websockets, curl_cffi
[debug] Loaded 1810 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp-nightly-builds/releases/latest
Latest version: nightly@2024.04.28.232723 from yt-dlp/yt-dlp-nightly-builds
yt-dlp is up to date (nightly@2024.04.28.232723 from yt-dlp/yt-dlp-nightly-builds)
[generic] Extracting URL: https://pmvhaven.com/video/MARAVILLOSA-PMV-_66195f01d0f2168854325fd0
[generic] MARAVILLOSA-PMV-_66195f01d0f2168854325fd0: Downloading webpage
WARNING: [generic] Falling back on generic information extractor
[generic] MARAVILLOSA-PMV-_66195f01d0f2168854325fd0: Extracting information
[debug] Looking for embeds
[debug] Identified a Open Graph video info
[debug] Formats sorted by: hasvid, ie_pref, lang, quality, res, fps, hdr:12(7), vcodec:vp9.2(10), channels, acodec, size, br, asr, proto, vext, aext, hasaud, source, id
[debug] Default format spec: bestvideo*+bestaudio/best
[info] MARAVILLOSA (PMV)_66195edfd0f2168854325d15: Downloading 1 format(s): 0
[debug] Invoking http downloader on "https://storage.pmvhaven.com/66195edfd0f2168854325d15/MARAVILLOSA (PMV)_66195edfd0f2168854325d15.mp4"
[debug] File locking is not supported. Proceeding without locking
[download] Destination: [pmvhaven.com] NA MARAVILLOSA (PMV) [MARAVILLOSA (PMV)_66195edfd0f2168854325d15].mp4
[download] 100% of  353.24MiB in 00:00:04 at 81.26MiB/s
```
```

**Comment by dirkf** [comment]
The page contains this `<meta name="video-id" content="66195f01d0f2168854325fd0">` which could be extracted by the Generic extractor.

The page metadata has no date, but the descriptions I checked were all like this `'Experience the mesmerizing PMV - {title} created by {uploader}'`: it should be possible to process this with yt-dlp options to remove such a boilerplate `description` generated by the Generic extractor and create an `uploader` instead.

For a dedicated extractor the interesting point is that the Nuxt.js hydration data doesn't have the expected function call structure, but is just plain JSON with some scheme where values are indirected via integers.

**Comment by hunkyburrito** [comment]
Would be cool to see support for this site now that the new site update (Nov. 2025) has broken generic extractor support :/

There's a scraper I found that may be useful for getting other metadata: 
https://github.com/stashapp/CommunityScrapers/blob/master/scrapers/PMVHaven/PMVHaven.py

**Comment by bashonly** [comment]
https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy
https://github.com/yt-dlp/yt-dlp/wiki/FAQ#what-does-sites-primarily-used-for-piracy-mean
https://github.com/yt-dlp/yt-dlp/wiki/FAQ#why-is-there-a-rule-against-websites-primarily-used-for-piracy

**Comment by hunkyburrito** [comment]
I disagree since the site also prohibits stolen and unedited copyrighted content.

https://pmvhaven.com/rules

Regardless, thanks for your work and consideration.

**Comment by m1karii** [comment]
@bashonly so what makes you think "the website is primarily used for piracy"?
from what i can tell the site hosts exclusively short form compilation style music videos from their original creators who cannot make any profit from them which makes the works both "transformative" and "non-commercial" according to fair use
